src/test/java/nwoolcan/model/brewery/warehouse/StockImplTest.java

```diff
@@ -0,0 +1,93 @@
+package nwoolcan.model.brewery.warehouse;
+
+import nwoolcan.model.brewery.warehouse.article.Article;
+import nwoolcan.model.brewery.warehouse.article.ArticleImpl;
+import nwoolcan.model.utils.Quantities;
+import nwoolcan.model.utils.Quantity;
+import nwoolcan.model.utils.UnitOfMeasure;
+import nwoolcan.utils.Empty;
+import nwoolcan.utils.Result;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+
+/**
+ * Test class for StockImpl.
+ */
+public class StockImplTest {
+
+    private Stock stock;
+    private Stock stock1;
+    private Date expDate;
+    private Record record1 = new Record(Quantity.of(TEN, UOM), Record.Action.ADDING);
+    private Record record2 = new Record(Quantity.of(ONE, UOM), Record.Action.REMOVING);
+    private Record record3 = new Record(Quantity.of(ONE, UOM1), Record.Action.ADDING);
+    private Record record4 = new Record(Quantity.of(TEN, UOM), Record.Action.REMOVING);
+    private Record record5 = new Record(Quantity.of(TEN, UOM), Record.Action.ADDING);
+    private static final Integer ID = 1;
+    private static final Integer ONE = 1;
+    private static final Integer TEN = 10;
+    private static final String NAME = "DummyName";
+    private static final String RECORD_WITH_DIFFERENT_UOM = "Can't add a record if UOMS are not the same";
+    private static final String REMOVING_RECORD_WITH_QUANTITY_NOT_AVAILABLE = "Can't add a remove"
+                                                       + "record if the quantity is not available";
+    private static final UnitOfMeasure UOM = UnitOfMeasure.Kilogram;
+    private static final UnitOfMeasure UOM1 = UnitOfMeasure.Liter;
+    private static final Article ARTICLE = new ArticleImpl(ID, NAME, UOM);
+
+    /**
+     * Initialize structures.
+     */
+    @Before
+    public void init() {
+        expDate = new Date();
+        stock = new StockImpl(ID, ARTICLE, expDate);
+        stock1 = new StockImpl(ID, ARTICLE);
+        try {
+            Thread.sleep(TEN);
+        } catch (Exception ex) {
```

Comment:
uops, there's a try-catch here...

Reply:
```suggestion
        Result.ofChecked(() -> Thread.sleep(TEN))
```

Reply:
to fix this, i need you to Add the version of Results.ofChecked() which accept void as return type of the lambda

Reply:
Done